lib/parsers/webp_parser.rb

```diff
@@ -0,0 +1,123 @@
+# WebP is an image format that provides superior lossless and lossy compression for images on the web, with support for
+# transparency. It uses predictive coding to encode an image, predicting the values in a block of pixels based on the
+# values of neighbouring blocks. A WebP file consists of VP8 or VP8L data, and a container based on RIFF. There is also
+# an extended file format, VP8X, that optionally encodes various information such as the color profile, animation
+# control data, transparency, and EXIF and/or XMP metadata.
+#
+# For more information, visit https://developers.google.com/speed/webp.
+#
+# TODO: Extract Exif and XMP metadata.
```

Comment:
Ps: if at the end there is work left for not releasing the full parsing of the format let's just remember to create the related issue on the repo itself, so that who wants can pick it up

Reply:
I would say give it a try and let's see where you get at. It would be nice to release full parser whenever possible! I understand it can be very cumbersome, I had this experience myself not long ago, but so far luckily this format doesn't look too complex!
What do you mean the whole file needs to be parsed? If for example this kind of format is the type of "content header says the next 600bytes are content and then we find the EXIF/XMP" then I'd definitely go for it.
If instead you have to go through 20 different internal headers in the content, and scrutinize each one of them for a chance of a nested EXIF, then it's another story. Let me know!